src/vcpkg-test/manifests.cpp

```diff
@@ -1263,3 +1263,216 @@ TEST_CASE ("license error messages", "[manifests][license]")
   on expression: MIT AND unknownlicense
                          ^)");
 }
+
+TEST_CASE ("default-feature-core errors", "[manifests]")
+{
+    auto m_pgh = test_parse_project_manifest(R"json({
+      "default-features": ["core"]
+    })json",
+                                             PrintErrors::No);
+    REQUIRE(!m_pgh.has_value());
+    REQUIRE(m_pgh.error()->to_string() == "error: while loading <test manifest>:\n"
+                                          "$.default-features[0] (a default feature): the feature \"core\" turns off "
+                                          "default features and thus can't be in the default features list");
+}
+
+TEST_CASE ("default-feature-core-object errors", "[manifests]")
+{
+    auto m_pgh = test_parse_project_manifest(R"json({
+      "default-features": [ { "name": "core" } ]
+    })json",
+                                             PrintErrors::No);
+    REQUIRE(!m_pgh.has_value());
+    REQUIRE(m_pgh.error()->to_string() == "error: while loading <test manifest>:\n"
+                                          "$.default-features[0].name (a default feature): the feature \"core\" turns "
+                                          "off default features and thus can't be in the default features list");
+}
+
+TEST_CASE ("default-feature-default errors", "[manifests]")
+{
+    auto m_pgh = test_parse_project_manifest(R"json({
+      "default-features": ["default"]
+    })json",
+                                             PrintErrors::No);
+    REQUIRE(!m_pgh.has_value());
+    REQUIRE(m_pgh.error()->to_string() ==
+            "error: while loading <test manifest>:\n"
+            "$.default-features[0] (a default feature): the feature \"default\" refers to "
+            "the set of default features and thus can't be in the default features list");
+}
+
+TEST_CASE ("default-feature-default-object errors", "[manifests]")
+{
+    auto m_pgh = test_parse_project_manifest(R"json({
+      "default-features": [ { "name": "default" } ]
+    })json",
+                                             PrintErrors::No);
+    REQUIRE(!m_pgh.has_value());
+    REQUIRE(m_pgh.error()->to_string() ==
+            "error: while loading <test manifest>:\n"
+            "$.default-features[0].name (a default feature): the feature \"default\" refers to the set of default "
+            "features and thus can't be in the default features list");
+}
+
+TEST_CASE ("default-feature-empty errors", "[manifests]")
+{
+    auto m_pgh = test_parse_project_manifest(R"json({
+      "default-features": [""]
+    })json",
+                                             PrintErrors::No);
+    REQUIRE(!m_pgh.has_value());
+    REQUIRE(m_pgh.error()->to_string() == "error: while loading <test manifest>:\n"
+                                          "$.default-features[0] (a feature name): \"\" is not a valid feature name. "
```

Comment:
If an error contains multiple sentences, I ended the first with a period and capitalize to start the next one.